src/Types/AccountAddress.cs

```diff
@@ -0,0 +1,163 @@
+using Concordium.Sdk.Client;
+using Concordium.Sdk.Helpers;
+using NBitcoin.DataEncoders;
+
+namespace Concordium.Sdk.Types;
+
+/// <summary>
+/// Represents an account address.
+///
+/// The address of an account is a 32-byte value which uniquely identifies the account.
+/// </summary>
+public readonly struct AccountAddress : IEquatable<AccountAddress>
+{
+    public const uint BytesLength = 32;
+
+    /// <summary>
+    /// A version byte that is prepended to addresses represented as base58 strings.
+    /// </summary>
+    private const byte VersionByte = 1;
+
+    private static readonly Base58CheckEncoder _encoderInstance = new();
+
+    /// <summary>
+    /// Representation of the account address as a byte array (without the version byte prepended).
+    /// </summary>
+    private readonly byte[] _value;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AccountAddress"/> class.
+    /// </summary>
+    /// <param name="addressAsBytes">The address as a sequence of bytes without a version byte prepended.</param>
+    private AccountAddress(byte[] addressAsBytes) => this._value = addressAsBytes;
+
+    /// <summary>
+    /// Get the address represented as a base58 encoded string.
+    /// </summary>
+    public override string ToString() =>
+        // Prepend version byte.
+        _encoderInstance.EncodeData((new byte[] { VersionByte }).Concat(this._value).ToArray());
+
+    /// <summary>
+    /// Get the address as a length-32 byte array without the version byte prepended.
+    /// </summary>
+    public byte[] GetBytes() => (byte[])this._value.Clone();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AccountAddress"/> class.
+    /// </summary>
+    /// <param name="addressAsBase58String">The address represented as a length-50 base58 encoded string.</param>
+    /// <exception cref="ArgumentException">The input is not a length-50 base58 encoded string.</exception>
+    public static AccountAddress From(string addressAsBase58String)
+    {
+        try
+        {
+            var decodedBytes = _encoderInstance.DecodeData(addressAsBase58String).Skip(1).ToArray(); // Remove version byte.
```

Comment:
Not that you need to do it now, but generally when you now the size of arrays and you are skipping, concatenating etc. you would like to work with Span.

Example if you instead wrote
```
            var decodeData = _encoderInstance.DecodeData(addressAsBase58String).AsSpan();
            var decodedBytes = decodeData[1..].ToArray(); // Remove version byte.
```

You would reduce memory allocation by 10 % in a small benchmark test.

```
|    Method |     Mean |     Error |    StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
|---------- |---------:|----------:|----------:|------:|--------:|-------:|----------:|------------:|
| UsingSkip | 2.425 us | 0.0110 us | 0.0302 us |  1.00 |    0.00 | 0.2708 |     568 B |        1.00 |
| UsingSpan | 2.324 us | 0.0082 us | 0.0227 us |  0.96 |    0.02 | 0.2480 |     520 B |        0.92 |
```

again - you don’t need to refactor now but now you know :-)

Reply:
Nice, thanks :-) might as well do that.